glue.py

```diff
@@ -64,10 +74,27 @@ def define(self, key, val):
         self.build_kwargs['defines'].append('{0}={1}'.format(key, val))
 
     def build(self, module, ctx):
-        self.build_kwargs['use'] = [m._name for m in module.depends]
-        self.build_kwargs['source'] = module.files
+        sources = []
+        objects = []
+
+        for fname in module.files:
+            if re.match('.*\.o', fname):
```

Comment:
Missing end anchor (otherwise `foo.oww` matches as well). Why not just `fname.endswith('.o')`?